Mage.Sets/src/mage/cards/f/FreestriderCommando.java

```diff
@@ -0,0 +1,78 @@
+package mage.cards.f;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.abilities.keyword.PlotAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.stack.Spell;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class FreestriderCommando extends CardImpl {
+
+    public FreestriderCommando(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}");
+
+        this.subtype.add(SubType.CENTAUR);
+        this.subtype.add(SubType.MERCENARY);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Freestrider Commando enters the battlefield with two +1/+1 counters on it if it wasn't cast or no mana was spent to cast it.
+        this.addAbility(new EntersBattlefieldAbility(
+                new ConditionalOneShotEffect(
+                        new AddCountersSourceEffect(CounterType.P1P1.createInstance(2)),
+                        FreestriderCommandoCondition.instance, ""
+                ), "with two +1/+1 counters on it if it wasn't cast or no mana was spent to cast it"
+        ));
+
+        // Plot {3}{G}
+        this.addAbility(new PlotAbility("{3}{G}"));
+    }
+
+    private FreestriderCommando(final FreestriderCommando card) {
+        super(card);
+    }
+
+    @Override
+    public FreestriderCommando copy() {
+        return new FreestriderCommando(this);
+    }
+}
+
+enum FreestriderCommandoCondition implements Condition {
+    instance;
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return false;
+        }
+        // Check if the spell exists on the stack
+        Spell spell = game.getStack().getSpell(source.getSourceId());
+        if (spell == null) {
+            return true; // cannot find the spell, so it wasn't cast.
```

Comment:
huh, is this reliable enough, no cast watcher needed?

Reply:
You actively do NOT want to use cost tags for this. Cost tags would be copied by [[Double Major]].

I'm pretty sure the current setup also fails with Double Major though, I think you do need a cast watcher.

Reply:
Both Commando & Satoru have been working properly with Double Major in my live tests.
I'll add unit tests though.
![image](https://github.com/magefree/mage/assets/34709007/3dbfc341-d233-4c62-a147-b7c064a2bd36)
![image](https://github.com/magefree/mage/assets/34709007/bc7e44a1-e5bb-479c-8b56-1a6a01349114)

Reply:
@Susucre why you ignore stack here, but check it in same conditional for SatoruTheInfiltrator?

Reply:
What do you mean ignore the stack?

Reply:
Both cards uses same conditional:

> it wasn't cast or no mana was spent to cast it

> none of them were cast or no mana was spent to cast them

First card check spell only, but second card: stack zone + spell.

![IMG_0150](https://github.com/magefree/mage/assets/8344157/985538c0-a2f5-4e54-8cd4-e26805e3ee9f)

Reply:
Well the stack zone is because Satoru is triggering on events, so it was just there to check only the permanents that were moved from the stack.
Check could probably be removed you're right, as if not from stack, getSpellOrLKIStack would return null.